File: agents/python/src/prompt.py
"""
Block 1: System Prompt & Persona (The Rules)

Loads the shared system prompt and prepares it with the project path.
Identical logic across all language implementations.
"""

import os

PROMPT_FILE = os.path.join(os.path.dirname(__file__), "..", "..", "spec", "system-prompt.txt")


def load_system_prompt(project_path: str) -> str:
    with open(PROMPT_FILE, "r") as f:
        template = f.read()
    return template.replace("{PROJECT_PATH}", project_path)


def get_user_prompt(project_path: str) -> str:
    return (
        f"The project at {project_path} has several bugs causing production errors. "
        f"Find and fix all of them. Start by examining the error logs in the logs/ directory."
    )

File: agents/javascript/src/prompt.ts
/**
 * Block 1: System Prompt & Persona (The Rules)
 *
 * Loads the shared system prompt and prepares it with the project path.
 * This is identical across all language implementations.
 */

import * as fs from 'fs';
import * as path from 'path';

const PROMPT_FILE = path.join(__dirname, '..', '..', 'spec', 'system-prompt.txt');

export function loadSystemPrompt(projectPath: string): string {
  const template = fs.readFileSync(PROMPT_FILE, 'utf-8');
  return template.replace(/\{PROJECT_PATH\}/g, projectPath);
}

export function getUserPrompt(projectPath: string): string {
  return `The project at ${projectPath} has several bugs causing production errors. Find and fix all of them. Start by examining the error logs in the logs/ directory.`;
}
